bagel/logger.py

```diff
@@ -1,45 +1,67 @@
 import logging
+import sys
+from enum import Enum
 from typing import NoReturn
 
 import typer
 from rich.logging import RichHandler
 
-from bagel.config import CONFIG
-
 LOG_FMT = "%(message)s"
 DATETIME_FMT = "[%Y-%m-%d %X]"
 
+# Check if code is currently running in a test environment
+IS_TESTING = "pytest" in sys.modules
+
+logger = logging.getLogger("bagel.logger")
+
+
+class VerbosityLevel(str, Enum):
+    """Enum for verbosity levels."""
+
+    ERROR = "0"
+    INFO = "1"
+    # NOTE: We currently do not have any debug logs, but can add them in the future
+    DEBUG = "2"
+
 
 # TODO: Once we introduce CLI options for reducing or increasing verbosity,
 # we can use the "level" parameter to set the logging level
-def get_logger(level: int = logging.INFO) -> logging.Logger:
-    """Create a logger with the specified logging level."""
-
-    logger = logging.getLogger("test")
-    handler = RichHandler(
-        omit_repeated_times=False, show_path=False, rich_tracebacks=True
-    )
-    formatter = logging.Formatter(fmt=LOG_FMT, datefmt=DATETIME_FMT)
+def configure_logger(level: int = logging.INFO):
+    """Configure a logger with the specified logging level."""
+
+    # Prevent duplicate handlers when updating the logger
+    if not logger.handlers:
+        handler = RichHandler(
+            omit_repeated_times=False, show_path=False, rich_tracebacks=True
+        )
+        formatter = logging.Formatter(fmt=LOG_FMT, datefmt=DATETIME_FMT)
+        handler.setFormatter(formatter)
+        logger.addHandler(handler)
+
     logger.setLevel(level)
-    handler.setLevel(level)
-    handler.setFormatter(formatter)
-    logger.addHandler(handler)
+    for handler in logger.handlers:
+        handler.setLevel(level)
 
     # propagate should be False to avoid duplicate messages from root logger
-    logger.propagate = False
-
-    return logger
+    # except when testing because otherwise Pytest does not capture the logs
+    if not IS_TESTING:
+        logger.propagate = False
 
 
 def log_error(
     logger: logging.Logger,
     message: str,
 ) -> NoReturn:
-    """Log an exception with or without the full traceback, and exit the app."""
-    # when exc_info=True, the current exception information will be fetched
-    # and included in the log after the custom message
-    logger.error(message, exc_info=CONFIG["debug"])
+    """Log an exception with an informative error message, and exit the app."""
+    logger.error(message)
     raise typer.Exit(code=1)
 
 
-logger = get_logger()
+def set_log_level(verbosity: VerbosityLevel):
+    """Set the logging level based on the verbosity option."""
+    if verbosity == VerbosityLevel.ERROR:
+        configure_logger(logging.ERROR)
+    elif verbosity == VerbosityLevel.INFO:
+        configure_logger(logging.INFO)
+    elif verbosity == VerbosityLevel.DEBUG:
+        configure_logger(logging.DEBUG)
```

Comment:
:cherries: could this be a dictionary lookup instead? It's pretty nice and explicit, but there is a lot of repetition. If we had

```python
myLevels = {
  VerbosityLevel.ERROR: logging.ERROR,
  VerbosityLevel.INFO: logging.INFO,
...
}
```
then I could just do:

```
configure_logger(myLevel.get(verbosity))
```